### PR TITLE
Add conversion stability tests

### PR DESCRIPTION
**Pull request by fedepaol**
Here we add a set of tests to validate that the conversion between two apis are stable and doesn't depend on the order.
We check:
- the metallb api -> internal config conversion
- the internal config -> frr configuration
- the internal config -> frr k8s configuration

### CODE REVIEW

**Comment by fedepaol** [comment]
The wip part is the tests revealing the conversion is not stable :P 
To be fixed.

File: internal/shuffler/shuffler_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+type TestStruct2 struct {
+	A int
+	B []string
+	C []TestStruct
+	D []*TestStruct
+}
+
+func TestShuffle(t *testing.T) {
```
Comment (oribon) [inline internal/shuffler/shuffler_test.go:38]: does this do anything?

File: internal/shuffler/shuffler_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	rnd := rand.New(rand.NewSource(23))
+	foo := []string{"a", "b", "c"}
+	Shuffle(foo, rnd)
+}
+
+func TestShuffleStruct(t *testing.T) {
```
Comment (oribon) [inline internal/shuffler/shuffler_test.go:44]: same

File: internal/bgp/frr/testdata/TestConversionIsStable.golden [hunk 3]
```diff
@@ -1,4 +1,4 @@
-log file /etc/frr/frr.log informational
```
Comment (oribon) [inline internal/bgp/frr/testdata/TestConversionIsStable.golden]: nit: belongs to a different commit

**Review by oribon — APPROVED** [review]
lgtm

File: internal/config/config_test.go [hunk 4]
```diff
@@ -3449,6 +3452,224 @@ func TestContainsAdvertisement(t *testing.T) {
...
+	seed := time.Now().UnixNano()
+	rnd := rand.New(rand.NewSource(seed))
+	var prev *Config
+	for i := 0; i < 10; i++ {
+		shuffler.Shuffle(toCheck, rnd)
+		got, err := For(toCheck, DontValidate)
+		if err != nil {
+			t.Fail()
+		}
+		if prev != nil && cmp.Equal(*got, *prev, cmpopts.IgnoreUnexported(Pool{})) {
```
Comment (oribon) [inline internal/config/config_test.go:3665]: why is this `cmp.Equal` (could you add a comment if it's intended) ?
